lightning/src/chain/keysinterface.rs

```diff
@@ -184,8 +192,12 @@ impl ChannelKeys for InMemoryChannelKeys {
 	fn htlc_base_key(&self) -> &SecretKey { &self.htlc_base_key }
 	fn commitment_seed(&self) -> &[u8; 32] { &self.commitment_seed }
 
-	fn sign_remote_commitment<T: secp256k1::Signing>(&self, channel_value_satoshis: u64, channel_funding_redeemscript: &Script, feerate_per_kw: u64, commitment_tx: &Transaction, keys: &TxCreationKeys, htlcs: &[&HTLCOutputInCommitment], to_self_delay: u16, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()> {
+	fn sign_remote_commitment<T: secp256k1::Signing>(&self, channel_value_satoshis: u64, feerate_per_kw: u64, commitment_tx: &Transaction, keys: &TxCreationKeys, htlcs: &[&HTLCOutputInCommitment], to_self_delay: u16, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()> {
 		if commitment_tx.input.len() != 1 { return Err(()); }
+
+		let funding_pubkey = PublicKey::from_secret_key(secp_ctx, &self.funding_key);
+		let channel_funding_redeemscript = make_funding_redeemscript(&funding_pubkey, &self.remote_funding_pubkey.expect("must set remote funding key before signing"));
```

Comment:
nit: I think &Option<X: Copy>.expect() will technically Copy the type (its kinda dumb that PubliKey implements Copy when its such a large type). LLVM should trivially catch this with optimization, but you probably want Option<X: Copy>.as_ref().expect().

Reply:
fixed, rebased, squashed